### PR TITLE
[issue-67] extract isselected method/snippet from mnemonic confirm

### PR DESCRIPTION
**Pull request by thatStrangeGuyThorchain**
- [x] moved isSelected to separate factory helper

### CODE REVIEW

File: src/components/wallet/NewMnemonicConfirm.tsx [hunk 1]
```diff
@@ -5,6 +5,8 @@ import { Card, Col, Row, Button, Form } from 'antd'
 import { Store } from 'antd/lib/form/interface'
 import { v4 as uuidv4 } from 'uuid'
 
+import { isSelectedFactory } from '../../helpers/isSelectedHelper'
```
Comment (veado) [inline src/components/wallet/NewMnemonicConfirm.tsx:8]: How about to name this helper just `isSelected` (w/o `factory`) and the module `helpers/arrayHelper` or `helpers/listHelper`?

File: src/components/wallet/NewMnemonicConfirm.tsx [hunk 2]
```diff
@@ -65,13 +67,8 @@ const MnemonicConfirmScreen: React.FC<{ mnemonic: string; onConfirm: Function }>
...
   init()
 
-  const isSelected = useCallback(
-    (id: string) => {
-      const res: WordType | undefined = wordsList.find((e: WordType) => e._id === id)
-      return !!res?.selected
-    },
-    [wordsList]
-  )
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (veado) [inline src/components/wallet/NewMnemonicConfirm.tsx:70]: Why do we need this ^ here?

Comment (thatStrangeGuyThorchain) [reply src/components/wallet/NewMnemonicConfirm.tsx:70]: react's preset linter can not infer returned type from factory function return type and throws a warning that deps list is unknown

File: src/components/wallet/NewMnemonicConfirm.tsx (hunk 1 above)
Comment (thatStrangeGuyThorchain) [reply src/components/wallet/NewMnemonicConfirm.tsx:8]: but.... this is a factory for isSelected functions😅

Comment (veado) [reply src/components/wallet/NewMnemonicConfirm.tsx:8]: Ah, I've missed that. `factory` is fine in this case ;)

**Pull request merged**